share/examples/bitcoin.conf

```diff
@@ -4,6 +4,10 @@
  
 # Network-related settings:
 
+# Note that if you use testnet or regtest, particularly with the options
```

Comment:
I think this is the best place to show sections (instead of pointing to the end of file) - after all, sections are for "Network related settings". This way you also avoid talking about "addnode, connect, port, bind, rpcport, rpcbind or wallet ..." twice.

Reply:
I'm onboard that the repetition is poor in English, but I think the alternatives are worse:

If the section explanation at the bottom is moved up, we move commentary away from its subject in the file.

If we fix that by moving the sections themselves with the commentary, it changes the meaning of the miscellaneous options such that uncommenting #keypool=100, which currently would affect all three networks, will afterwards only affect regtest.

If we fix that by copy/pasting the options from regtest to each section then we're much increasing the size of a file creating a maintenance burden.

If we instead move #keypool=100 onwards above the network-related settings then we've a configuration file that unhappily starts with 'Miscellaneous options'.

What might work is to move the section explanation up and follow it with a [general] section, but I don't believe that's supported, and I'm reluctant to fake it with a '# [general]' pseudo-section that attempts to ties the hands of future developers.

What might also work is to have multiple configuration examples rather than or as well as this one attempting to be all things to all people. That would be a direction for the future rather than for this PR.

bitcoin.conf is in a mess and my repetition doesn't help but it does feel like the least intrusive way to get reference to sections into the example.

Reply:
> it occurs to me that section terminators might also work: [/main] etc, however I don't believe they're supported either.

Right, it's not implemented.